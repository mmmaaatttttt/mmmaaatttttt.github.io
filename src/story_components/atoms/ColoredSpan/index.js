import styled from "styled-components";
import PropTypes from "prop-types";

const ColoredSpan = styled.span`
  color: ${props => props.color};
  font-weight: ${props => props.bold ? "bold" : "normal"};
`;

ColoredSpan.propTypes = {
  bold: PropTypes.bool.isRequired,
  color: PropTypes.string.isRequired
};

ColoredSpan.defaultProps = {
  bold: false,
  color: "black"
};

export default ColoredSpan;
