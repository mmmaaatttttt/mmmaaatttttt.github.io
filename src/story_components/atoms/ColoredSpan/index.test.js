import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import ColoredSpan from ".";

it("renders successfully", () => {
  shallow(<ColoredSpan />);
});

it("matches snapshot", () => {
  const wrapper = shallow(
    <ColoredSpan bold color="blue">
      i'm blue, da bo dee da bo dah
    </ColoredSpan>
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
