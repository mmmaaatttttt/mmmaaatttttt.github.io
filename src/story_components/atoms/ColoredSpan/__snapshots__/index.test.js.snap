// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<StyledComponent
  bold={true}
  color="blue"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  color: ",
          [Function],
          ";
  font-weight: ",
          [Function],
          ";
",
        ],
      },
      "defaultProps": Object {
        "bold": false,
        "color": "black",
      },
      "displayName": "styled.span",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "bold": [Function],
        "color": [Function],
      },
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "span",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  i'm blue, da bo dee da bo dah
</StyledComponent>
`;
